File: QuantumHost/Programme.cs
// See https://aka.ms/new-console-template for more information
using Microsoft.Quantum.Simulation.Simulators;
using QuantumArithmetic;

Console.WriteLine("Hello, World!");
using var quantumSimulator = new QuantumSimulator();

var plus = await quantumSimulator.Run<Add, (double, double), double>((3, 4));
Console.WriteLine($"3 + 4 = {plus}");

var minus = await quantumSimulator.Run<Subtract, (double, double), double>((10, 7));
Console.WriteLine($"10 - 7 = {minus}");

var product = await quantumSimulator.Run<Multiply, (double, double), double>((5, 6));
Console.WriteLine($"5 * 6 = {product}");

var shareBetween = await quantumSimulator.Run<Divide, (double, double), double>((12, 3));
Console.WriteLine($"12 / 3 = {shareBetween}");

do
{
    Console.WriteLine("Please enter a number (or type exit to quit): ");
    var valueOne = Console.ReadLine();

    if (!string.IsNullOrWhiteSpace(valueOne) && valueOne.ToLower().Equals("exit"))
        break;

    Console.WriteLine("Please enter an operator (+, -, x, /, or type exit to quit): ");
    var valueOperator = Console.ReadLine();

    if (!string.IsNullOrWhiteSpace(valueOperator) && valueOperator.ToLower().Equals("exit"))
        break;

    Console.WriteLine("Please enter a number (or type exit to quit): ");
    var valueTwo = Console.ReadLine();

    if (!string.IsNullOrWhiteSpace(valueTwo) && valueTwo.ToLower().Equals("exit"))
        break;

    if (!double.TryParse(valueOne, out var x) || !double.TryParse(valueTwo, out var y))
    {
        Console.WriteLine("invalid entry for values!");
        return;
    }
    else if (string.IsNullOrWhiteSpace(valueOperator))
    {
        Console.WriteLine("Invalid operator!");
        return;
    }
    else
    {
        double output = 0.0;

        switch (valueOperator)
        {
            case "+":
                output = await quantumSimulator.Run<Add, (double, double), double>((x, y));
                Console.WriteLine(output);
                break;
            case "-":
                output = await quantumSimulator.Run<Subtract, (double, double), double>((x, y));
                Console.WriteLine(output);
                break;
            case "x":
                output = await quantumSimulator.Run<Multiply, (double, double), double>((x, y));
                Console.WriteLine(output);
                break;
            case "/":
                output = await quantumSimulator.Run<Divide, (double, double), double>((x, y));
                Console.WriteLine(output);
                break;
            default:
                Console.WriteLine("Invalid operator!");
                return;
        }
    }
}
while (true);

File: QuantumHost/Programme.vb
Imports Microsoft.Quantum.Simulation.Simulators
Imports QuantumArithmetic
Module Module1

    Sub Main()
        'See https://aka.ms/New-console-template for more information

        Console.WriteLine("Hello, World!")
        Using quantumSimulator As New QuantumSimulator

            Dim plus As Double = Await quantumSimulator.Run(Of Add, (Double, Double), Double)((3, 4))
Console.WriteLine($"3 + 4 = {plus}")

            Dim minus As Double = Await quantumSimulator.Run(Of Subtract, (Double, Double), Double)((10, 7))
Console.WriteLine($"10 - 7 = {minus}")

            Dim product As Double = Await quantumSimulator.Run(Of Multiply, (Double, Double), Double)((5, 6))
Console.WriteLine($"5 * 6 = {product}")

            Dim shareBetween As Double = Await quantumSimulator.Run(Of Divide, (Double, Double), Double)((12, 3))
Console.WriteLine($"12 / 3 = {shareBetween}")

            Do While True

                Console.WriteLine("Please enter a number (or type exit to quit): ")
                Dim valueOne As String = Console.ReadLine()

                If (Not String.IsNullOrWhiteSpace(valueOne) AndAlso valueOne.ToLower().Equals("exit")) Then
                    Exit While

                    Console.WriteLine("Please enter an operator (+, -, x, /, or type exit to quit): ")
                    Dim valueOperator As String = Console.ReadLine()

                    If (Not String.IsNullOrWhiteSpace(valueOperator) AndAlso valueOperator.ToLower().Equals("exit")) Then
                        Exit While

                        Console.WriteLine("Please enter a number (or type exit to quit): ")
                        Dim valueTwo As String = Console.ReadLine()

                        If (Not String.IsNullOrWhiteSpace(valueTwo) AndAlso valueTwo.ToLower().Equals("exit")) Then
                            Exit While

                            Dim x, y As Double
                            If (Not Double.TryParse(valueOne, x) OrElse Not Double.TryParse(valueTwo, y)) Then
                                Console.WriteLine("invalid entry for values!")
                                Continue While
                            ElseIf (String.IsNullOrWhiteSpace(valueOperator)) Then
                                Console.WriteLine("Invalid operator!")
                                Continue While
                            Else
                                Dim output As Double = 0.0

                                Select Case (valueOperator)
                                    Case "+"
                                        output = Await quantumSimulator.Run(Of Add, (Double, Double), Double)((x, y))
                                        Console.WriteLine(output)
                                    Case "-"
                                        output = Await quantumSimulator.Run(Of Subtract, (Double, Double), Double)((x, y))
                                        Console.WriteLine(output)
                                    Case "x"
                                        output = Await quantumSimulator.Run(Of Multiply, (Double, Double), Double)((x, y))
                                        Console.WriteLine(output)
                                    Case "/"
                                        output = Await quantumSimulator.Run(Of Divide, (Double, Double), Double)((x, y))
                                        Console.WriteLine(output)
                                    Case Else
                                        Console.WriteLine("Invalid operator!")
                                End Select

                            End If

                        End If
                    End If
                End If
            Loop
        End Using
    End Sub

End Module